launcher/ui/MainWindow.cpp

```diff
@@ -139,785 +140,101 @@ QString profileInUseFilter(const QString & profile, bool used)
-        connect(instanceToolBar, &QToolBar::orientationChanged, [=](Qt::Orientation){ instanceToolBar->setOrientation(Qt::Vertical); });
-        instanceToolBar->setAllowedAreas(Qt::LeftToolBarArea | Qt::RightToolBarArea);
-        instanceToolBar->setToolButtonStyle(Qt::ToolButtonTextBesideIcon);
-        instanceToolBar->setIconSize(QSize(16, 16));
-
-        instanceToolBar->setFloatable(false);
-        instanceToolBar->setWindowTitle(QT_TRANSLATE_NOOP("MainWindow", "Instance Toolbar"));
-
-        instanceToolBar->addWidget(changeIconButton);
-        instanceToolBar->addWidget(renameButton);
-
-        instanceToolBar->addSeparator();
-
-        instanceToolBar->addAction(actionLaunchInstance);
-        instanceToolBar->addAction(actionKillInstance);
-
-        instanceToolBar->addSeparator();
+        ui->instanceToolBar->insertSeparator(ui->actionLaunchInstance);
 
-        instanceToolBar->addAction(actionEditInstance);
-        instanceToolBar->addAction(actionChangeInstGroup);
-
-        instanceToolBar->addAction(actionViewSelectedInstFolder);
+        // restore the instance toolbar settings
+        auto const setting_name = QString("WideBarVisibility_%1").arg(ui->instanceToolBar->objectName());
+        if (!APPLICATION->settings()->contains(setting_name))
+            instanceToolbarSetting = APPLICATION->settings()->registerSetting(setting_name);
+        else
+            instanceToolbarSetting = APPLICATION->settings()->getSetting(setting_name);
 
-        instanceToolBar->addAction(actionExportInstance);
-        instanceToolBar->addAction(actionCopyInstance);
-        instanceToolBar->addAction(actionDeleteInstance);
+        ui->instanceToolBar->setVisibilityState(instanceToolbarSetting->get().toByteArray());
 
-        instanceToolBar->addAction(actionCreateInstanceShortcut); // TODO find better position for this
+        ui->instanceToolBar->addContextMenuAction(ui->newsToolBar->toggleViewAction());
+        ui->instanceToolBar->addContextMenuAction(ui->instanceToolBar->toggleViewAction());
+        ui->instanceToolBar->addContextMenuAction(ui->actionLockToolbars);
 
-        QLayout * lay = instanceToolBar->layout();
-        for(int i = 0; i < lay->count(); i++)
-        {
-            QLayoutItem * item = lay->itemAt(i);
-            if (item->widget()->metaObject()->className() == QString("QToolButton"))
-            {
-                item->setAlignment(Qt::AlignLeft);
-            }
-        }
-
-        all_toolbars.append(&instanceToolBar);
-        MainWindow->addToolBar(Qt::RightToolBarArea, instanceToolBar);
     }
 
-    void setupUi(MainWindow *MainWindow)
+    // set the menu for the folders and help tool buttons
     {
-        if (MainWindow->objectName().isEmpty())
-        {
-            MainWindow->setObjectName(QStringLiteral("MainWindow"));
-        }
-        MainWindow->resize(800, 600);
-        MainWindow->setWindowIcon(APPLICATION->getThemedIcon("logo"));
```

Comment:
This seems to have been removed without a replacement, the window icon is currently missing on builds from this pr